### ISSUE TITLE
Inactive role seem to install transition steps

### ISSUE THREAD

**Issue by Zelldon** [camunda-cloud/zeebe#7896]
**Description**

When runnnig test on develop where a broker is stopped we can see in the logs that the broker transitions to inactive in the end, it looks like that the transition steps are installed

```
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 starting
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing LogStorage
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing LogStream
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing ZeebeDb
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing QueryService
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing StreamProcessor
08:41:42.597 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing SnapshotDirector
08:41:42.598 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 - executing ExporterDirector
08:41:42.598 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] INFO  io.camunda.zeebe.broker.system - Transition to INACTIVE on term -1 completed
```

I saw code that this is checked on each transition step, which feels a bit error prone and also confusing in regards to reading the log. I would prefer that we not install any steps on inactive step.

**Comment by npepinpe** [comment]
@deepthidevaki - afaik this is only a logging issue, and we don't actually install anything, right?

**Comment by Zelldon** [comment]
https://github.com/camunda-cloud/zeebe/blob/develop/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/LogStoragePartitionTransitionStep.java#L45-L46

This check needs to be done on all transition steps, but we could just avoid to install the step in the first place.

**Comment by deepthidevaki** [comment]
It is not "installing" anything. It is "Transitioning" to inactive which should close all open service.

**Comment by Zelldon** [comment]
What should tell me this then: `Transition to INACTIVE on term -1 - executing LogStorage`

**Comment by deepthidevaki** [comment]
> What should tell me this then: `Transition to INACTIVE on term -1 - executing LogStorage`

LogStorage is transitioning to inactive mode.

**Comment by npepinpe** [comment]
Where is the -1 coming from?

**Comment by Zelldon** [comment]
From here https://github.com/camunda-cloud/zeebe/blob/develop/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/NewPartitionTransitionImpl.java#L25

**Comment by npepinpe** [comment]
Can we improve this so I'm the last person to ask this? :smile:

**Comment by Zelldon** [comment]
Depends what the out come of this issue is :laughing:

**Comment by npepinpe** [comment]
Easy improvements: if we're transitioning to inactive, just omit the term. I know it's ugly to include a branch for logging, but the -1 does look a little confusing. If we really don't want, we can slightly reword it to have that INACTIVE ignores the term and always outputs -1.

I find the rest OK imo - to clarify, it could be `applying step STEP_NAME` instead of just `executing STEP_NAME`, but I think that's not such a big improvement. Maybe what Zell wants more is a separation between rollback and installing? So it's clearer the that inactive only performs rollback, and no install. But at the same time, this distinction may not always be true, so I'm not sure it's all that helpful. imo I would leave it as is, but if more people find it confusing I'm happy to accept alternatives. Otherwise let's leave it as is.

**Comment by Zelldon** [comment]
> I find the rest OK imo - to clarify, it could be applying step STEP_NAME instead of just executing STEP_NAME, but I think that's not such a big improvement. Maybe what Zell wants more is a separation between rollback and installing? So it's clearer the that inactive only performs rollback, and no install. But at the same time, this distinction may not always be true, so I'm not sure it's all that helpful. imo I would leave it as is, but if more people find it confusing I'm happy to accept alternatives. Otherwise let's leave it as is.

Unclear is to me why I need it. I remove/uninstall anyway all steps on transitioning. Why iterate overall of them again with inactive role and nothing happens. Doesn't make sense to me.

**Comment by npepinpe** [comment]
I don't want to speak for Deepthi and Peter, they may have had other ideas, but for me the big one is debugging/observability. If something goes wrong in the middle, I know where it went wrong, what was the last step performed successfully, etc. If it hangs, or times out, same - you know up until which step it got.

**Comment by deepthidevaki** [comment]
> I remove/uninstall anyway all steps on transitioning

This is not true. We remove or uninstall steps conditionally. For example, if we transition from follower to leader, we do not re-install ZeebeDb.

**Comment by deepthidevaki** [comment]
> Easy improvements: if we're transitioning to inactive, just omit the term. 

Can we use the current term (term from the previous role)?

**Comment by npepinpe** [comment]
Most likely? I almost suggested it, but then I figured there's the slight edge case where you start and haven't seen an election anyway, so you might still have -1? :man_shrugging:

**Comment by deepthidevaki** [comment]
> Most likely? I almost suggested it, but then I figured there's the slight edge case where you start and haven't seen an election anyway, so you might still have -1? man_shrugging

I think on startup the term will be either 0 or 1. :thinking:

**Comment by npepinpe** [comment]
I guess it's fine, and it's actually better in the common case, so we could just do that.

**Comment by npepinpe** [comment]
Let's update the message to use the last known term in inactive instead of -1.

**Comment by npepinpe** [comment]
Let's do this now as it should be pretty fast. @deepthidevaki can you do it? Let me know if it turns out it's not a very quick fix :smile:

**Comment by deepthidevaki** [comment]
To use the last term in inactive role? Yes, I can do it.
